### PR TITLE
Transfer identities to search api

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

This is to send Identifier collection to searchApi

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

tested locally and with unit test.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: FAMS-738

### CODE REVIEW

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        public string Identification { get; set; }
+
+        [Description("The identification effective date.")]
+        public DateTime? IdentificationEffectiveDate { get; set; }
+
+        [Description("The identification expiration date.")]
+        public DateTime? IdentificationExpirationDate { get; set; }
+
+        [Description("The identfication type,such as driver license.")]
+        public int IdentifierType { get; set; }
```
Comment (alexjoybc) [inline app/SearchApi/SearchApi.Web/People/Identifier.cs]: this should be an enum or a class, the int is insufficient to resolve the identifier

File: app/SearchApi/SearchApi.Web/People/PersonSearchRequest.cs [hunk 2]
```diff
@@ -25,5 +27,6 @@ public PersonSearchRequest(string firstName, string lastName, DateTime? dateOfBi
         public string LastName { get; }
         [Description("The date of birth of the subject.")]
         public DateTime? DateOfBirth { get; }
+        public List<Identifier> Identifiers { get; set; }
```
Comment (alexjoybc) [inline app/SearchApi/SearchApi.Web/People/PersonSearchRequest.cs]: Can we use IEnumerable ?

File: app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/PersonSearchRequestBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public class PersonSearchRequestBuilder
+    {
+        private PersonSearchRequest _personSearchRequest;
+
+        public PersonSearchRequestBuilder()
+        {
+        }
+
+        public PersonSearchRequestBuilder WithSearchApiRequest(SSG_SearchApiRequest request)
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/SearchRequest/PersonSearchRequestBuilder.cs]: The builder pattern works the other way around, you want to expose a method that set a property of the object like firstName lastName...., the builder pattern helps with reducing the amount of parameters in the constructor and separate the logic of object initialization
```c#
public PersonSearchRequestBuilder WithFirstName(string firstName) { 
...
}
```
what you are looking for here is to map a SSG_SearchApiRequest to a PersonSearchRequest, a mapping class would suit better, or you can look at [automapper](https://automapper.org/), maybe for latter

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SearchApi.Web.People
+{
+    [Description("Represents a set of information to execute a search on a person")]
+    public class Identifier
+    {
+        [JsonConstructor]
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: ok. will remove.

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        public Identifier()
+        {
+        }
+
+        [Description("The identification serial number of this identifier.")]
+        public string Identification { get; set; }
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: how about "SerialNumber"?

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+        [Description("The identification effective date.")]
+        public DateTime? IdentificationEffectiveDate { get; set; }
+
+        [Description("The identification expiration date.")]
+        public DateTime? IdentificationExpirationDate { get; set; }
+
+        [Description("The identfication type,such as driver license.")]
+        public int IdentifierType { get; set; }
+
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: ok. good catch.

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        [Description("The inforation source of this identifier, such as icbc.")]
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: removed.

File: app/SearchApi/SearchApi.Web/People/Identifier.cs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        public int InformationSourceType{ get; set; }
```
[the remark this answers is not in the document]
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: ok. change name to Source.

File: app/SearchApi/SearchApi.Web/People/Identifier.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/Identifier.cs]: agree

File: app/SearchApi/SearchApi.Web/People/PersonSearchRequest.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/SearchApi/SearchApi.Web/People/PersonSearchRequest.cs]: ok.

**Pull request merged**